src/sentry/search/events/builder.py

```diff
@@ -1165,7 +1180,7 @@ def _default_filter_converter(self, search_filter: SearchFilter) -> Optional[Whe
         if (
             search_filter.operator in ("!=", "NOT IN")
             and not search_filter.key.is_tag
-            and name != "event.type"
+            and name not in self._non_nullable_keys
```

Comment:
Can we move this to the dataset config?
```suggestion
            and name not in self.config.non_nullable_keys
```